cookbooks/bcpc-hadoop/recipes/maven_config.rb

```diff
@@ -2,14 +2,15 @@
 
 node.override['maven']['install_java'] = false
 
-internet_download_url = node['maven']['3']['url']
-maven_file = Pathname.new(node['maven']['3']['url']).basename
+internet_download_url = node['maven']['url']
+maven_file = Pathname.new(node['maven']['url']).basename
 
-node.override['maven']['3']['url'] = "#{get_binary_server_url}/#{maven_file}"
+node.override['maven']['url'] = "#{get_binary_server_url}/#{maven_file}"
 
 # download Maven only if not already stashed in the bins directory
 remote_file "/home/vagrant/chef-bcpc/bins/#{maven_file}" do
   source internet_download_url
-  action :create_if_missing
-  mode '0555'
+  action :create
```

Comment:
I added a checksum using the attribute provided by the maven cookbook.

If the file exists, and the checksum fits, it won't reach out to the internet.
If the file doesn't match the checksum, grumpiness is desired behavior.